src/GitHub.Api/Git/RepositoryManager.cs

```diff
@@ -9,18 +9,18 @@ namespace GitHub.Unity
     interface IRepositoryManager : IDisposable
     {
         event Action<bool> OnIsBusyChanged;
-
-        event Action OnCommitChanged;
-        event Action OnLocalBranchListChanged;
-        event Action OnRemoteBranchListChanged;
+        event Action<string> OnHeadUpdated;
         event Action<GitStatus> OnStatusUpdated;
-        event Action<ConfigBranch?> OnActiveBranchChanged;
-        event Action<ConfigRemote?> OnActiveRemoteChanged;
+        event Action<IEnumerable<GitLock>> OnLocksUpdated;
+        event Action<Dictionary<string, ConfigBranch>> OnLocalBranchListUpdated;
+        event Action<Dictionary<string, Dictionary<string, ConfigBranch>>> OnRemoteBranchListUpdated;
```

Comment:
The `RepositoryManager` will instead communicate new information via "Updated" events.